File: cdk-from-cfn-testing/expected/cloudwatch/java/src/main/java/com/myorg/CloudwatchConstruct.java
package com.myorg;

import software.constructs.Construct;

import java.util.*;
import software.amazon.awscdk.CfnMapping;
import software.amazon.awscdk.CfnTag;
import software.amazon.awscdk.Stack;
import software.amazon.awscdk.StackProps;

import software.amazon.awscdk.*;
import software.amazon.awscdk.services.cloudwatch.*;

class CloudwatchConstruct extends Construct {
    public CloudwatchConstruct(final Construct scope, final String id) {
        this(scope, id, null);
    }

    public CloudwatchConstruct(final Construct scope, final String id,
            String environmentName) {
        super(scope, id);

        environmentName = Optional.ofNullable(environmentName).isPresent() ? environmentName
                : "dev";

        CfnAlarm myApi5xxErrorsAlarm = CfnAlarm.Builder.create(this, "MyApi5xxErrorsAlarm")
                .alarmDescription("Example alarm")
                .namespace("AWS/ApiGateway")
                .dimensions(Arrays.asList(
                        CfnAlarm.DimensionProperty.builder()
                                .name("ApiName")
                                .value("MyApi")
                                .build()))
                .metricName("5XXError")
                .comparisonOperator("GreaterThanThreshold")
                .statistic("Average")
                .threshold(0.005)
                .period(900)
                .evaluationPeriods(1)
                .treatMissingData("notBreaching")
                .alarmActions(Arrays.asList(
                        Fn.importValue(environmentName + "AlarmsTopicArn")))
                .build();

    }
}

File: cdk-from-cfn-testing/expected/cloudwatch/golang/CloudwatchConstruct.go
package main

import (
	"fmt"

	cdk "github.com/aws/aws-cdk-go/awscdk/v2"
	cloudwatch "github.com/aws/aws-cdk-go/awscdk/v2/awscloudwatch"
	"github.com/aws/constructs-go/constructs/v10"
	"github.com/aws/jsii-runtime-go"
)

type CloudwatchConstructProps struct {
	/// Environment used for this deployment.
	EnvironmentName *string
}

type CloudwatchConstruct struct {
	constructs.Construct
}

func NewCloudwatchConstruct(scope constructs.Construct, id string, props *CloudwatchConstructProps) *CloudwatchConstruct {
	construct := constructs.NewConstruct(scope, &id)

	cloud_watch.NewCfnAlarm(
		construct,
		jsii.String("MyApi5xxErrorsAlarm"),
		&cloud_watch.CfnAlarmProps{
			AlarmDescription: jsii.String("Example alarm"),
			Namespace: jsii.String("AWS/ApiGateway"),
			Dimensions: &[]interface{}{
				&DimensionProperty{
					Name: jsii.String("ApiName"),
					Value: jsii.String("MyApi"),
				},
			},
			MetricName: jsii.String("5XXError"),
			ComparisonOperator: jsii.String("GreaterThanThreshold"),
			Statistic: jsii.String("Average"),
			Threshold: jsii.Number(0.005),
			Period: jsii.Number(900),
			EvaluationPeriods: jsii.Number(1),
			TreatMissingData: jsii.String("notBreaching"),
			AlarmActions: &[]*string{
				cdk.Fn_ImportValue(jsii.String(fmt.Sprintf("%vAlarmsTopicArn", props.EnvironmentName))),
			},
		},
	)

	return &CloudwatchConstruct{
		Construct: construct,
	}
}

